### PR TITLE
ci: add pep8-naming check to flake8 and ignore a bunch of warnings

### PR DESCRIPTION
**Pull request by jon-hagg**
### Purpose
Add pep8-naming check to flake8, so it is run by tox locally and in github. 
See https://github.com/Breakthrough-Energy/RenewableEnergyProject/issues/343

### What the code is doing
In this case, we ignore almost all the warnings. Note: adding the noqa on some lines also caused black to reformat into multi line declarations, which in turn requires moving the noqa back to the first line in that group. Hopefully we don't have to do this a lot. 

### Testing
`tox -e flake8`

### Time estimate
5 min

### CODE REVIEW

**Review by BainanXia — APPROVED** [review]
Looks good. Thanks for taking care of this.

File: powersimdata/design/transmission/tests/test_upgrade.py [hunk 1]
```diff
@@ -139,43 +139,43 @@ def setUp(self):
         self.grid.id2zone = {201: "Wahington", 202: "Oregon"}
         self.grid.zone2id = {"Washington": 201, "Oregon": 202}
 
-    def test_internal_W(self):
+    def test_internal_W(self):  # noqa: N802
```
Comment (danielolsen) [inline powersimdata/design/transmission/tests/test_upgrade.py:142]: These function names are an artifact of old tests where we had dummy zones `"E"` and `"W"`. Now that we've changed the tests to `Oregon`/`Washington`, it probably makes sense to change the test names.

File: powersimdata/design/investment/create_mapping_files.py [hunk 2]
```diff
@@ -223,7 +223,7 @@ def write_poly_shapefile():
     """
     fiona = _check_import("fiona")
     shapely_geometry = _check_import("shapely.geometry")
-    Polygon = shapely_geometry.Polygon
+    polygon = shapely_geometry.Polygon
```
Comment (danielolsen) [inline powersimdata/design/investment/create_mapping_files.py:226]: Should we keep this as `Polygon` and `#noqa` the error? What we're really doing is importing a class, which I think we should keep uppercase if possible.

Comment (jon-hagg) [reply powersimdata/design/investment/create_mapping_files.py:226]: Sure, I think it makes sense in this case.

File: powersimdata/design/transmission/tests/test_upgrade.py (hunk 1 above)
Comment (jon-hagg) [reply powersimdata/design/transmission/tests/test_upgrade.py]: I gave it a shot, let me know if it makes sense.

**Comment by danielolsen** [comment]
@lanesmith can we come up with some more descriptive variable names for the cost curve visualizations?

**Comment by lanesmith** [comment]
> @lanesmith can we come up with some more descriptive variable names for the cost curve visualizations?

@danielolsen, yeah certainly. What's the best way for me make those additions to this PR? Should I just checkout this branch?

**Comment by danielolsen** [comment]
> @danielolsen, yeah certainly. What's the best way for me make those additions to this PR? Should I just checkout this branch?

I suggest that you check out this branch, create a new branch of your own with your commit(s), and push that. Then @jon-hagg can cherry-pick your commits into this branch when ready, and there's no risk of conflicts.

**Comment by lanesmith** [comment]
@jon-hagg, I've finished updating the variable names for the cost curve visualization. The updates are in the branch `lane/cost_curves_refactor`. Let me know how you would like to proceed with this.

**Comment by jon-hagg** [comment]
> @jon-hagg, I've finished updating the variable names for the cost curve visualization. The updates are in the branch `lane/cost_curves_refactor`. Let me know how you would like to proceed with this.

Thanks @lanesmith, probably the easiest way would be to create a PR to this branch, which (when merged) will effectively append your commits to the current PR.

**Review by danielolsen — APPROVED** [review]
Looks good, thanks!

**Pull request merged**